lib/cpp/src/thrift/transport/TPipeServer.cpp

```diff
@@ -95,9 +96,15 @@ class TAnonPipeServer : public TPipeServerImpl {
 
 class TNamedPipeServer : public TPipeServerImpl {
 public:
-  TNamedPipeServer(const std::string& pipename, uint32_t bufsize, uint32_t maxconnections)
-    : stopping_(false), pipename_(pipename), bufsize_(bufsize), maxconns_(maxconnections)
-  {
+  TNamedPipeServer(const std::string& pipename,
+                   uint32_t bufsize,
+                   uint32_t maxconnections,
+                   const std::string& securityDescriptor)
```

Comment:
> to protect against this in Thrift it should allow the security of a pipe to be configured

This forces the issue, and is a breaking change.  It would be better to introduce this functionality while still allowing existing code to compile, and deprecate the old constructors, and then remove them in the following release if we really think it is necessary to force a securityDescriptor in all use cases (I'm not convinced yet).

Reply:
I agree that this shouldn't be a breaking a change and shouldn't enforce the use of a security descriptor. TNamedPipeServer is internal to TPipeServer. I have given TPipeServer an additional constructor that now takes a security descriptor as a parameter. Now when TPipeServer is constructed unless a security descriptor is provided it has the security descriptor set to "Everyone" to maintain the previous behavior . As TNamedPipeServer  is only in the cpp I don't see how this is a breaking change.